pandas/core/reshape/melt.py

```diff
@@ -417,9 +417,7 @@ def wide_to_long(
     """
 
     def get_var_names(df, stub: str, sep: str, suffix: str) -> List[str]:
-        regex = r"^{stub}{sep}{suffix}$".format(
-            stub=re.escape(stub), sep=re.escape(sep), suffix=suffix
-        )
+        regex = r"^{re.escape(stub)}{re.escape(sep)}{suffix}$"
```

Comment:
```suggestion
        regex = fr"^{re.escape(stub)}{re.escape(sep)}{suffix}$"
```

Missing the prefix here